lib/provisioning/Provisioner.js

```diff
@@ -128,60 +129,48 @@ function Provisioner(ircBridge, enabled, requestTimeoutSeconds) {
         next();
     });
 
-    as.app.post("/_matrix/provision/link", Promise.coroutine(function*(req, res) {
-        try {
-            yield self.requestLink(req.body);
-            res.json({});
-        }
-        catch (err) {
-            res.status(500).json({error: err.message});
-            throw err;
-        }
-    }));
+    let createProvisionEndpoint = (fn, fnName) => {
+        return Promise.coroutine(function*(req, res) {
+            req = new ProvisionRequest(req, fnName);
+            req.log.info(
+                'New provisioning request: ' + JSON.stringify(req.body) +
+                ' params: ' + JSON.stringify(req.params)
+            );
+            try {
+                let result = yield fn.apply(self, [req]);
+                if (!result) {
+                    result = {};
+                }
+                req.log.info(`Sending result: ${JSON.stringify(result)}`);
+                res.json(result);
```

Comment:
When the time comes when we do want some non-standard error codes, I'll be happy to create some way of sticking that in a result object. The twitter bridge does it by returning `{err: 404, body: "some message about not finding something"}`, and then scraping `err` to give to `res.status` and `body` to give to `res.json`.